test/shared/src/main/scala/zio/test/mock/MockClock.scala

```diff
@@ -26,6 +26,47 @@ import zio.internal.{ Scheduler => IScheduler }
 import zio.internal.Scheduler.CancelToken
 import zio.scheduler.Scheduler
 
+/**
+ * `MockClock` makes it easy to deterministically and efficiently test effects
+ * involving the passage of time.
+ *
+ * Instead of waiting for actual time to pass, `sleep` and methods implemented
+ * in terms of it schedule effects to take place at a given time. Users can
+ * adjust the current clock time using the `adjust` and `setTime` methods, and
+ * all effects scheduled to take place on or before that time will automically
+ * be run.
+ *
+ * For example, here is how we can test [[ZIO.timeout]] using `MockClock:
+ *
+ * {{{
+ *  import zio.ZIO
+ *  import zio.duration._
+ *  import zio.test.mock.MockClock
+ *
+ *  for {
+ *    mockClock <- MockClock.makeMock(MockClock.DefaultData)
+ *    fiber <- ZIO.sleep(5.minutes).timeout(1.minute).fork
+ *    _ <- mockClock.setTime(1.minute)
+ *    result <- fiber.join
+ *   } yield result == None
+ * }}}
+ *
+ * Note how we forked the fiber that `sleep` was invoked on. Calls to `sleep`
+ * and methods derived from it will semantically block until the time is
+ * set to on or after the time they are scheduled to run. If we didn't fork the
+ * fiber on which we called sleep we would never get to set the the time on the
+ * line below. Thus, a useful pattern when using `MockClock` is to fork the
+ * effect being tested, then adjust the clock to the desired time, and finally
+ * verify that the expected effects have been performed.
+ *
+ * One other thing to keep in mind is that `sleep` and related combinators
+ * schedule events to occur at an absolute point in time (e.g. 10 seconds from
+ * the `0` time), not relative time (e.g. 10 seconds from the current set clock
+ * time). This is necessary to avoid race conditions between scheduling effects
+ * and adjusting the time but means it sometimes may be necessary to use
+ * `adjustTime` to reset the time between effects you are testing to achieve
```

Comment:
would be great to have an example here showing this use case

Reply:
Good suggestion. I will work on something and submit a follow up PR.